### DISCUSSION TITLE
Lost 2FA

### DISCUSSION THREAD

**Discussion by ryanoirl** [community/community#67933] [New to GitHub]
Hi,

   I had my recovery codes for 2FA authentication on an old machine that I no longer have. Is there anyway to reset 2FA on a github account? I had a repos in there I want to get access to config.

Thanks

**Comment by kuroodo** [comment]
If you still have your authenticator, maybe try contacting support.

If you lost your authenticator AND your recovery codes, then you've lost your account. The only thing support is willing to do is detach the email so you can make a new account with the same email.

Welcome to the 2FA experience! This is going to be mandatory in the entire platform next month. Go use GitLab or other competitors instead.

**Comment by melvincwng** [comment]
@ryanoirl, hopefully, this link below can provide some helpful solutions to allow you to recover your account.

GitHub Documentation Reference: [Recovering your account if you lose your 2FA credentials](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/recovering-your-account-if-you-lose-your-2fa-credentials)

Hope this helps, take care and hope you manage to recover your account 🙏🏻

**Comment by tuliogontijo** [comment]
I'm in the same situation! Lost my device and don't have my recovery codes. I tried to contact a human by chat, mail, phone, but found no way. I talk to a virtual assistant, but in the end it tells me to unlink my e-mail and fork the repositories I wanted. But I don't want it. There are lots of private repositories that I need and won't be able to fork.

Can anybody give me a solution?

**Reply by kuroodo** [reply]
If you still have other forms of authentication that give you access to these repos via git, you can still fork your repos through there so that you don't lose them.


Unfortunately the only solution is to learn from your mistake and be extremely careful with 2FA. In my opinion 2FA is a nightmare. I highly recommend you move to GitLab. They have 2FA through email.

**Reply by tuliogontijo** [reply]
It is likely that on a computer at my work I still have access to the CLI. I think I can do pulls and pushes there. Could you tell me if it is possible to do something in this CLI to recover my account?

**Reply by kuroodo** [reply]
> if it is possible to do something in this CLI to recover my account?

You'll have to make a ticket to support and ask them. It can take at least a week for support to respond sometimes.

Until then, make sure to rescue your private repos through the CLI ASAP
